syntax = "proto3";
package Thesis;

service DataStreamHandler{
  rpc exchange_silent_data(stream Data) returns (stream Result) {}
  rpc exchange_overt_data(Data) returns (Param) {}
  rpc shutdown(Param) returns (Param) {}
}

message Param{}

message Data{
  bytes value = 1;
  string type = 2;
  string label = 3;
  MetaData headset_information = 4;
}

message MetaData{
  int32 channels = 1;
  int32 number_of_samples = 2;
  int32 participant_id = 3;
}

message Result{
  string value = 1;
}
